src/Features/CSharp/Portable/Completion/CompletionProviders/SymbolCompletionProvider.cs

```diff
@@ -259,5 +262,66 @@ private static CompletionItemRules MakeRule(bool importDirective, bool preselect
         }
 
         protected override CompletionItemSelectionBehavior PreselectedItemSelectionBehavior => CompletionItemSelectionBehavior.HardSelection;
+
+        protected override CompletionItem CreateItem(
+            CompletionContext completionContext,
+            string displayText,
+            string displayTextSuffix,
+            string insertionText,
+            List<ISymbol> symbols,
+            SyntaxContext context,
+            bool preselect,
+            SupportedPlatformData supportedPlatformData)
+        {
+            var item = base.CreateItem(
+                completionContext,
+                displayText,
+                displayTextSuffix,
+                insertionText,
+                symbols,
+                context,
+                preselect,
+                supportedPlatformData);
+
+            var isInferredTypeDelegate = context.InferredTypes.Any(type => type.IsDelegateType());
+            var isObjectCreationTypeContext = context switch
+            {
+                CSharpSyntaxContext csharpSyntaxContext => csharpSyntaxContext.IsObjectCreationTypeContext,
+                _ => false
+            };
+
+            var shouldProvideParenthesisCompletion = symbols.All(symbol => symbol switch
+            {
+                IMethodSymbol => !isInferredTypeDelegate,
+                ITypeSymbol => isObjectCreationTypeContext,
+                IAliasSymbol => isObjectCreationTypeContext,
+                _ => false
+            });
+
+            if (!shouldProvideParenthesisCompletion)
+            {
+                return item;
+            }
+
+            item = SymbolCompletionItem.AddProvideParenthesisCompletion(item, true);
+            var hasParameter = symbols.All(ShouldPutCaretBetweenParenthesis);
```

Comment:
This would cause compiler to bind mermbers of all symbols, which might be expensive. @CyrusNajmabadi does
`ReuseExistingSpeculativeModelAsync` above help address this issue?

Reply:
I now changed this to be checked when GetChangeAsync() is called. (By encoding the symbol into it)
So it should not cause perf problem now